org.ekstep.reviewercomments-1.0/editor/plugin.js

```diff
@@ -0,0 +1,131 @@
+                    var instance = this;
+                    var pkgVersion = ecEditor.getService('content').getContentMeta(org.ekstep.contenteditor.api.getContext('contentId')).pkgVersion;
+                    var data = {
+                        "request": {
+                            "contextDetails": {
+                                'contentId': ecEditor.getContext('contentId'),
+                                'contentType': 'application/vnd.ekstep.ecml-archive',
+                                'contentVer': !_.isUndefined(pkgVersion) ? pkgVersion.toString() : '0'
+                            }
+                        }
+                    };
+                    //Do Api call and get the response  
+                    ecEditor.getService(ServiceConstants.CONTENT_SERVICE).getComments(data, function (error, response) {
+                        if (!error) {
+                            //Loop through the response and generate the comments Thread
+                            // instance.comments = response.data.result.comments
+                            instance.showComments(response.data.result.comments);
+                        } else {
+                            instance.displayNoComments();
+                            ecEditor.dispatchEvent("org.ekstep.toaster:error", {
+                                message: "Error in fetching comments, please try again!",
+                                position: 'topCenter',
+                                icon: 'fa fa-warning'
+                            });
+                            ecEditor.getService('telemetry').error({
+                                "env": 'content',
+                                "stage": '',
+                                "action": 'review comments',
+                                "objectid": "",
+                                "objecttype": "",
+                                "err": error.status,
+                                "type": "API",
+                                "data": error,
+                                "severity": "fatal"
+                            })
+                        }
+                    });
+                }
+            }
+        } else {
+            instance.displayStageComments(instance.comments)
+        }
+    },
+    //Show comments after getting the apiResponse
+    showComments: function (comments) {
+        var instance = this;
+        var sortedComments = {}
+        if (comments !== undefined && comments.length > 0) {
+            sortedComments = _.sortBy(comments, function (dateObj) {
+                return new Date(dateObj.createdOn);
+            });
+            instance.mapComment(sortedComments);
+        }
+    },
+    //To show the 'No review message to the user
+    displayNoComments: function () {
+        jQuery('a[data-content ="comments"]').removeClass('highlight');
+        jQuery('#reviewerComments').html('<div>' + this.NO_COMMENTS + '</div>');
```

Comment:
Hardcoding Identifier `reviewerComments ` is danger, what if some other plugin is using same id 'reviewerComments'?